### DISCUSSION TITLE
Updating capsule state in listeners

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#99] [Q&A]
Hi again Greg.

This is about the issue I had with listeners. I was thinking that restricting capsules rebuilds to be triggered inside a listener is a problem in POV. EG you can call an abstract method inside a listener and you don't know whether it will try to change any capsule or not.

I quickly thought on a possible workaround and I want your opinion: what about modify or create another CapsuleContainer listen function that has as parameter a set of capsules and a callback function, then wrapping the callback with a closure that uses all that capsules and call the callback with a future.delayed

**Comment by busslina** [comment]
Yeah, I see a problem here. Some listeners want not only to be notified on a change but also to use the new value.

**Reply by busslina** [reply]
I see your current approach is the more flexible one and maybe with that warning that you will do is enough

**Reply by busslina** [reply]
But at the same time removing by default that restriction could have some benefits

**Comment by busslina** [comment]
Forget about. It make no sense

**Reply by GregoryConrad** [reply]
Well, you can always use `await null;` or `Future.delayed` in your own code. I wouldn't suggest it, unless to get something quickly working before refactoring for a proper fix later, because the root problem is likely elsewhere with how you're laying the capsules out.
